Realm/ObjectServerTests/SwiftObjectServerTests.swift

```diff
@@ -3079,6 +2223,124 @@ class AsyncAwaitObjectServerTests: SwiftSyncTestCase {
             }
         }
     }
+
+    func testCallResetPasswordAsyncAwait() async throws {
+        let email = "realm_tests_do_autoverify\(randomString(7))@\(randomString(7)).com"
+        let password = randomString(10)
+        try await app.emailPasswordAuth.registerUser(email: email, password: password)
+        do {
+            try await app.emailPasswordAuth.callResetPasswordFunction(email: email,
+                                                                      password: randomString(10),
+                                                                      args: [[:]])
+        } catch {
+            XCTAssertNotNil(error)
```

Comment:
I'm not sure what this test is trying to check. It appears to pass regardless of if callResetPasswordFunction succeeds or fails as long as the failure isn't a nil error, which seems to be a weird thing to be testing for?

Reply:
As mentioned on the internal slack channel testing this casuistic will need to implement handler that intercepts the email from the server and it is a lot of work, because of this we are testing at least for this to fail. I'm adding a  XCTFail() in case of success.